openfe/setup/ligand_network_planning.py

```diff
@@ -72,18 +76,37 @@ def generate_radial_network(ligands: Iterable[SmallMoleculeComponent],
 
 
 def minimal_spanning_graph(ligands: Iterable[SmallMoleculeComponent],
-                           mappers: Iterable[LigandAtomMapper], scorer=None):
+                           mappers: Iterable[LigandAtomMapper],
+                           scorer: Callable[[LigandAtomMapping], float]):
     """Plan a Network which connects all ligands with minimal cost
 
     Parameters
     ----------
-    ligands : Iterable of rdkit Molecules
+    ligands : Iterable[SmallMoleculeComponent]
       the ligands to include in the Network
-    mappers : Iterable of AtomMappers
+    mappers : Iterable[LigandAtomMapper]
       the AtomMappers to use to propose mappings.  At least 1 required,
       but many can be given, in which case all will be tried to find the
       lowest score edges
     scorer : Scoring function
-      any callable which takes an LigandAtomMapping and returns a float
+      any callable which takes a LigandAtomMapping and returns a float
     """
-    raise NotImplementedError
+    nodes = list(ligands)
+    mappings: List[LigandAtomMapping] = []  # for mypy 🙄
+
+    # First create a network with all the proposed mappings (scored)
+    mappings = sum([list(mapper.suggest_mappings(molA, molB))
```

Comment:
style quibble, keep this as a generator expression and evaluate it in the next step?